src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -1907,13 +1909,41 @@ static apt_bool_t speech_on_channel_add(mrcp_application_t *application, mrcp_se
 
 	/* notify of channel open */
 	if (globals.enable_profile_events && switch_event_create_subclass(&event, SWITCH_EVENT_CUSTOM, MY_EVENT_PROFILE_OPEN) == SWITCH_STATUS_SUCCESS) {
+		orig_session = switch_core_session_locate(schannel->session_uuid);
+		if (orig_session && switch_core_session_read_lock(orig_session) == SWITCH_STATUS_SUCCESS) {
+			orig_channel = switch_core_session_get_channel(orig_session);
+		}
 		switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "MRCP-Profile", schannel->profile->name);
 		if (schannel->type == SPEECH_CHANNEL_SYNTHESIZER) {
 			switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "MRCP-Resource-Type", "TTS");
 		} else {
 			switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "MRCP-Resource-Type", "ASR");
 		}
+		if (!zstr(schannel->unimrcp_session->id.buf)){
+			switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "MRCP-Session-Identifier", schannel->unimrcp_session->id.buf);
+		}
+		if (!zstr(schannel->unimrcp_session->signaling_agent->sip_id)) {
+			switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "MRCP-Session-SIP-Id", schannel->unimrcp_session->signaling_agent->sip_id);
+		}
+		if(orig_channel){
```

Comment:
In order for the call control to access this info, we need the flag `call_control` to be included in the channel variable. Or why not just use `switch_channel_event_set_data(orig_channel, event);` instead?

Reply:
I already have `call_control` flag set in line 1935. I didn't what to add all the channel event data to this event since majority of the data won't be needed.

Reply:
Miss that, since you have associated the telnyx uuid, you may also need to set the FS uuid in this event.

Reply:
Added